packages/NodeTypeResolver/PHPStan/Scope/Contract/NodeVisitor/NodeVisitorInterface.php

```diff
@@ -0,0 +1,11 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\NodeTypeResolver\PHPStan\Scope\Contract\NodeVisitor;
+
+use PhpParser\NodeVisitor;
+
+interface NodeVisitorInterface extends NodeVisitor
+{
+}
```

Comment:
empty interface act as a marking purpose, so instead it's better to use attributes IMO, and as the container is a Symfony one, it should be quite easy to use `AutoconfigureTag` and `TaggedIterator` attributes

Reply:
It extends `NodeVisitor` so it actually not empty.

I don't know how symfony works 😅

This is just way I see the current architecture, eg: `FileProcessorInterface`, `NodeTypeResolverInterface`, so it just a matter of consistency.

Reply:
it's not Symfony framework but more the Dependency Injection Symfony component.
anyway, I was just sharing. it's sure that using attributes this way will be rejected.

just in case, injection of the services based on phpdoc works thanks to `\Rector\Core\DependencyInjection\CompilerPass\AutowireArrayParameterCompilerPass`

Reply:
I would prefer using plain php instead of symfony specific features

Reply:
it's more about using the container features

Reply:
I prefer the "just works" of the current architecture instead of introduced another way to works with similar service definition, that make it inconsistent

Reply:
up to you, as said, just sharing :) showing the relatively "modern" way of marking classes thanks to attributes
of course consistency wins